File: binarysearch/Run-Length-Decoding/Run-Length-Decoding.cpp
// https://helloacm.com/c-run-length-decoding-algorithm/
// https://binarysearch.com/problems/Run-Length-Decoding
// EASY, STRING

string solve(string s) {
    int i = 0;
    string ans = "";
    while (i < s.size()) {
        int j = i;
        while (isdigit(s[j]) && (j < s.size())) {
            j ++;
        }
        int a = stoi(s.substr(i, j - i + 1));
        int k = j;
        while (isalpha(s[k]) && k < s.size()) {
            k ++;
        }
        auto t = s.substr(j, k - j);
        for (int k = 0; k < a; ++ k) {
            ans += t;
        }
        i = k;
    }    
    return ans;
}

File: binarysearch/Run-Length-Decoding/Run-Length-Decoding.py
# https://binarysearch.com/problems/Run-Length-Decoding
# https://helloacm.com/teaching-kids-programming-run-length-decoding-decompression-algorithm/
# EASY, STRING

class Solution:
    def solve(self, s):
        c = 0
        data = []
        for i in s:
            if i.isdigit():
                c = c * 10 + ord(i) - 48
            else:
                data.append(c)
                data.append(i)
                c = 0
        return "".join(data[i] * data[i + 1] for i in range(0, len(data), 2))
